gobblin-temporal/src/main/java/org/apache/gobblin/temporal/joblauncher/GobblinJobLauncher.java

```diff
@@ -206,6 +206,13 @@ public void launchJob(@Nullable JobListener jobListener) throws JobException {
         cancelJob(jobListener);
       }
     } finally {
+      // NOTE: This code only makes sense when there is 1 source / workflow being launched per application. This is a stop-gap
+      // for recreating batch job behavior in GaaS. Given the current constraints of yarn applications requiring a static proxy user
+      // during application creation, it is not possible to have multiple workflows running in the same application.
```

Comment:
regarding what I just said about inheritance, this might be a situation where that is in fact preferable to sticking opinionated behavior into a base class.

e.g. no more than a hook here, like `handleLaunchFinalization()`.  then create a derived class like `RunAndDoneGobblinJL` or `TerminateAfterLaunchGJL` that sends the cluster shutdown request.  then it's up to the specific job launchers to either take on this functionality or go w/o it by deciding which to extend.

Reply:
Im sticking it into the GobblinTemporalJobLauncher. Any deriving flow can override this default behavior if they want.